qualtran/bloqs/basic_gates/toffoli.py

```diff
@@ -59,6 +61,40 @@ def build_call_graph(self, ssa: 'SympySymbolAllocator') -> Set['BloqCountT']:
     def t_complexity(self):
         return TComplexity(t=4)
 
+    def add_my_tensors(
+        self,
+        tn: 'qtn.TensorNetwork',
+        tag: Any,
+        *,
+        incoming: Dict[str, 'SoquetT'],
+        outgoing: Dict[str, 'SoquetT'],
+    ):
+        import quimb.tensor as qtn
+
+        from qualtran.bloqs.basic_gates.cnot import XOR
+
+        # Set up the CTRL tensor which copies inputs to outputs and activates
+        # when a==1 and b==1
+        internal = qtn.rand_uuid()
+        inds = (
+            incoming['ctrl'][0],
+            incoming['ctrl'][1],
+            outgoing['ctrl'][0],
+            outgoing['ctrl'][1],
+            internal,
+        )
+        CTRL = np.zeros((2,) * 5, dtype=np.complex128)
+        for a, b in itertools.product([0, 1], repeat=2):
+            CTRL[a, b, a, b, int(a == 1 and b == 1)] = 1.0
+
+        # Wire up the CTRL tensor to XOR to flip `target` when active.
```

Comment:
Is there an obvious reason why this is called CTRL here and the analogue is COPY for CNOT add_my_tensors?

Reply:
You could use the (generalized) COPY tensor with three n-bit indices to implement e.g. unary iteration. 

```
COPY[a, a, a] = 1 for a in range(1,10) else 0
target[a, ...] = list_of_unitaries[a] for a in range(1, 10)
```

Reply:
interesting